lib/graphql/schema/loader.rb

```diff
@@ -102,7 +102,7 @@ def define_type(type, type_resolver)
               name: type["name"],
               type: type_resolver.call(type["type"]),
               description: type["description"],
-              default_value: type["defaultValue"]
+              default_value: type["defaultValue"] ? JSON.parse(type["defaultValue"], quirks_mode: true) : nil
```

Comment:
Is this what fixes the Enum dump + load bug? If so, can we get a test for this change? For example, I think it could be added to `loader_spec.rb`, you could: 

- update the enum into have a value whose Ruby representation is something custom
- add an argument and use that value as a default value
- then improve `assert_deep_equal` to check for `Argument#default_value` equality